kube-runtime/src/scheduler.rs

```diff
@@ -78,24 +78,24 @@ impl<T: Hash + Eq + Clone, R> SchedulerProj<'_, T, R> {
             .run_at
             .checked_add(*self.debounce)
             .unwrap_or_else(far_future);
-        match self.scheduled.entry(request.message) {
+        match self.scheduled.raw_entry_mut().from_key(&request.message) {
             // If new request is supposed to be earlier than the current entry's scheduled
             // time (for eg: the new request is user triggered and the current entry is the
             // reconciler's usual retry), then give priority to the new request.
-            Entry::Occupied(mut old_entry) if old_entry.get().run_at >= request.run_at => {
+            RawEntryMut::Occupied(mut old_entry) if old_entry.get().run_at >= request.run_at => {
                 // Old entry will run after the new request, so replace it..
                 let entry = old_entry.get_mut();
                 self.queue.reset_at(&entry.queue_key, next_time);
                 entry.run_at = next_time;
-                old_entry.replace_key();
+                old_entry.insert_key(request.message);
```

Comment:
This seems like the thing to pay attention to. AFAIKT this is re-inserting the same key and relying on key-depuplication in the map values to replace the entry?

I do see there's also something like [get_key_value_mut](https://docs.rs/hashbrown/latest/hashbrown/hash_map/struct.HashMap.html#method.get_key_value_mut) which perhaps makes this less ambiguous?

Reply:
`insert_key(key)` replaces the actual key of the entry with the passed key via `mem::replace()`, which should have the same effect as the former `replace_key()` in this instance.

AFAICT `get_key_value_mut()` provides only mutable access to the value and shared access to the key.

Reply:
Ah, fair, then this seems sensible then. 👍